src/items/associated-items.md

```diff
@@ -205,22 +205,47 @@ types cannot be defined in [inherent implementations] nor can they be given a
 default implementation in traits.
 
 An *associated type declaration* declares a signature for associated type
-definitions. It is written as `type`, then an [identifier], and
-finally an optional list of trait bounds.
+definitions. It is written in one of the following forms, where `Assoc` is the
+name of the associated type, `Params` is a comma-separated list of type,
+lifetime or const parameters, `Bounds` is a plus-separated list of trait bounds
+on the associated type, and `WhereBounds` is a comma-separated list of bounds on
+parameters:
+
+```rust,ignore
+type Assoc;
+type Assoc: Bounds;
+type Assoc<Params>;
+type Assoc<Params>: Bounds;
```

Comment:
It occurs to me that it would be worth clarifying the distinction between `Bounds` and `WhereBounds`!

Reply:
That is stated in the paragraph above? Should we be more detailed?

Reply:
I think we could be more explicit. Left some suggestions. WDYT?